test/publish.test.js

```diff
@@ -176,6 +176,133 @@ test.serial('Publish a release with an asset with a template label', async (t) =
   t.true(gitlab.isDone());
 });
 
+test.serial('Publish a release (with an link) with a template url', async (t) => {
+  const cwd = 'test/fixtures/files';
+  const owner = 'test_user';
+  const repo = 'test_repo';
+  const env = {GITLAB_TOKEN: 'gitlab_token'};
+  const nextRelease = {gitHead: '123', gitTag: 'v1.0.0', notes: 'Test release note body', version: '1.0.0'};
+  const options = {repositoryUrl: `https://gitlab.com/${owner}/${repo}.git`};
+  const encodedRepoId = encodeURIComponent(`${owner}/${repo}`);
+  const encodedGitTag = encodeURIComponent(nextRelease.gitTag);
+  const assets = [
+    {
+      label: 'README-v1.0.0.md',
+      type: 'other',
+      url: `https://gitlab.com/gitlab-org/gitlab/-/blob/master/README-v\${nextRelease.version}.md`,
+    },
+  ];
+  const gitlab = authenticate(env)
+    .post(`/projects/${encodedRepoId}/releases`, {
+      tag_name: nextRelease.gitTag,
+      description: nextRelease.notes,
+      assets: {
+        links: [
+          {
+            name: 'README-v1.0.0.md',
+            url: `https://gitlab.com/gitlab-org/gitlab/-/blob/master/README-v1.0.0.md`,
+            link_type: 'other',
+          },
+        ],
+      },
+    })
+    .reply(200);
+
+  const result = await publish({assets}, {env, cwd, options, nextRelease, logger: t.context.logger});
+
+  t.is(result.url, `https://gitlab.com/${owner}/${repo}/-/releases/${encodedGitTag}`);
+  t.deepEqual(t.context.log.args[0], ['Published GitLab release: %s', nextRelease.gitTag]);
+  t.true(gitlab.isDone());
+});
+
+test.serial('Publish a release (with an link) with a template type', async (t) => {
+  process.env.TYPE = 'other';
```

Comment:
Should we reset this at some point so that it does not interfere with other tests?

Reply:
Just now, I dont think there will be any interference, but better safe than sorry!👍🏼

Reply:
I reset the variables in line 237 & 238:
```
delete process.env.TYPE;
delete process.env.FILEPATH;
```